packages/eslint-plugin/src/rules/dependency-checks.ts

```diff
@@ -0,0 +1,363 @@
+      globalThis.projPackageJsonDeps;
+    const rootPackageJsonDeps = getAllDependencies(
+      join(workspaceRoot, 'package.json')
+    );
+
+    function validateMissingDependencies(node: AST.JSONProperty) {
+      if (!checkMissingDependencies) {
+        return;
+      }
+      const missingDeps = projDependencyNames.filter(
+        (d) => !projPackageJsonDeps[d]
+      );
+
+      missingDeps.forEach((d) => {
+        if (!ignoredDependencies.includes(d)) {
+          context.report({
+            node: node as any,
+            messageId: 'missingDependency',
+            data: { packageName: d, projectName: sourceProject.name },
+            fix(fixer) {
+              projPackageJsonDeps[d] =
+                rootPackageJsonDeps[d] || projDependencies[d];
+
+              const deps = (node.value as AST.JSONObjectExpression).properties;
+              if (deps.length) {
+                return fixer.insertTextAfter(
+                  deps[deps.length - 1] as any,
+                  `,\n    "${d}": "${projPackageJsonDeps[d]}"`
+                );
+              } else {
+                return fixer.replaceTextRange(
+                  [node.value.range[0] + 1, node.value.range[1] - 1],
+                  `\n    "${d}": "${projPackageJsonDeps[d]}"\n  `
+                );
+              }
+            },
+          });
+        }
+      });
+    }
+
+    function validateVersionMismatch(
+      node: AST.JSONProperty,
+      packageName: string,
+      packageRange: string
+    ) {
+      if (
+        !checkVersionMismatches ||
+        ignoredDependencies.includes(packageName) ||
+        satisfies(projDependencies[packageName], packageRange)
+      ) {
+        return;
+      }
+
+      context.report({
+        node: node as any,
+        messageId: 'versionMismatch',
+        data: {
+          packageName: packageName,
+          projectName: sourceProject.name,
+          version: projDependencies[packageName],
+        },
+        fix: (fixer) =>
+          fixer.replaceText(
+            node as any,
+            `"${packageName}": "${
+              rootPackageJsonDeps[packageName] || projDependencies[packageName]
+            }"`
+          ),
+      });
+    }
+
+    function validateObsoleteDependencies(
+      node: AST.JSONProperty,
+      packageName: string
+    ) {
+      if (
+        !checkObsoleteDependencies ||
+        ignoredDependencies.includes(packageName)
+      ) {
+        return;
+      }
+
+      context.report({
+        node: node as any,
+        messageId: 'obsoleteDependency',
+        data: { packageName: packageName, projectName: sourceProject.name },
+        fix: (fixer) => {
+          let isLastProperty = false;
+          if (
+            node.parent.properties[node.parent.properties.length - 1] === node
+          ) {
+            isLastProperty = true;
+          }
+          // remove 4 spaces, new line and potential comma from previous line
```

Comment:
We shouldn't assume that they have 4 spaces as indentation. Only remove the property and a trailing comma